### PR TITLE
Changed redirect on save from current page to surveys list

### PR DESCRIPTION
**Pull request by ryanchristo**
This pull request makes the following changes:
- changes redirect on save from current page to surveys list
- refactors save functions to ensure all saves are complete before redirecting

Testing checklist:
- [x] After I have successfully saved a survey, I am redirected to the survey list.
- [x] I am only redirected after roles, tasks and attributes are all successfully saved.

Fixes ushahidi/platform#1713.
Fixes ushahidi/platform#1750.

Ping @ushahidi/platform

### CODE REVIEW

File: app/settings/surveys/survey-editor.directive.js [hunk 1]
```diff
@@ -637,7 +637,7 @@ function SurveyEditorController(
         .saveCache(_.extend({ roles: $scope.roles_allowed }, { formId: $scope.survey.id }))
         .$promise
         .then(function (roles) {
-            $location.path('settings/surveys/edit/' + $scope.survey.id);
+            $location.url('/settings/surveys');
```
Comment (rjmackay) [inline app/settings/surveys/survey-editor.directive.js]: This makes sense. Theres an existing issue with both this and #676, that we're not waiting for *all* the requests to finish before redirecting, ie. roles might finish saving, but attributes + tasks are still processing, yet we we redirect anyway.

Do you think you could return promises from `saveRoles` `saveTasks` and `saveAttributes` then handle the redirect in `saveSurvey` ?
That enables catching any errors during save and only redirecting if *all* of them succeed.

Comment (ryanchristo) [inline app/settings/surveys/survey-editor.directive.js]: Yea, I can work on that. Should I start a new branch or keep working with this one?

Comment (ryanchristo) [inline app/settings/surveys/survey-editor.directive.js]: Yea, I can do that.

**Comment by ryanchristo** [comment]
@rjmackay Updated with refactored code to ensure all saves are complete before redirect.

**Comment by ryanchristo** [comment]
@rjmackay Updated with refactored code to ensure all saves are complete before redirect.

**Comment by ryanchristo** [comment]
@rjmackay Updated with refactored code to ensure all saves are complete before redirect.

File: app/settings/surveys/survey-editor.directive.js [hunk 2]
```diff
@@ -552,94 +552,93 @@ function SurveyEditorController(
...
-            saveTasks();
-            saveRoles();
-        }, handleResponseErrors);
-        $scope.saving_survey = false;
+            // If the survey is new, cache the new id
+            if ($scope.survey.id !== survey.id) {
+                $scope.survey.id = survey.id;
+            }
+            // Save tasks and return promises
+            return saveTasks();
```
Comment (rjmackay) [inline app/settings/surveys/survey-editor.directive.js]: You should still be able to run these in parallel, using `return $q.all([saveTasks(), saveRoles()]);` ... maybe including saveAttributes too.. I'm not sure if that relies on tasks saving first?

Comment (ryanchristo) [inline app/settings/surveys/survey-editor.directive.js]: Yea, I can fix that. `saveAttributes` relies on `saveTasks` but `saveRoles` can be run in parallel with `saveTasks`.

**Comment by ryanchristo** [comment]
@rjmackay Updated.

**Pull request merged**